docs/element-definitions.md

```diff
@@ -1270,9 +1270,12 @@ transitions.
     - `metadata` ([Array][][[API Metadata](#api-metadata-member)]) - Arbitrary metadata
 - `content` (array)
 
+
 #### Classifications
 
 - `"api"` - Category is a API top-level group.
+    - Metadata specific to related API document like `FORMAT`, `HOST` or document `version`. It SHOULD be placed
+    inside of attribute `metadata` as member element. Typical usecase is eg. [Metadata section](https://apiblueprint.org/documentation/specification.html#def-metadata-section) for Blueprint, or [Info object](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#info-object) for OpenAPI
```

Comment:
I don't get why this is needed to be said here. @kylef What are your thoughts?

Reply:
Do you have some proposal where to place this info? Or it is restful at all by our opinion?

Reply:
I am wondering if version should be placed inside the api metadata. Metadata is a concept originally from API Blueprint where it is arbitary user-defined values where we do not define the semantics of what the values may be in API Elements.

The version isn't an aribtary user define semantic, it is something with solid definition and we are treating it as such in the Swagger parser where we validate the value etc and tooling authors would expect that behaviour.

Since metadata isn't strict, it is down to the user/tooling to define the semantics and validation around them which I think is different to the version. I was thinking that the `version` would be an attribute of the api category. However I am open to this idea but I am weary that we will end up in a place where there can be parser discrepencies. If we place version in this metadata, tooling will expect it to follow the definition we defined, in other adapters/tools/users it could have other semantics.

If a user placed a version in API Blueprint as user-metadata then it wouldn't follow the same semantics and validation. Then tooling/users may file an issue saying that version isn't handled the same or the API Blueprint isn't validating that the version is in the correct format etc. I think we made this mistake with https://github.com/apiaryio/drafter/issues/481 although HOST is deployment information and something we didn't want to put into API Elements (at least at that time) so it remains custom user metadata.

Reply:
In general I can agree with predicate Metadata should be reserved for Blueprint and our tooling.

On other side. We do not write about it in spec.

So what about place swagger `info object` into same place as `attribure.info` with similar structure cause it has similar meaning.

it mean:

```
{
  "element": "category",
  "meta": {
    "classes": {
      "element": "array",
      "content": [
        {
          "element": "string",
          "content": "api"
        }
      ]
    },
  },
  "attributes": {
    "info": {
      "element": "array",
      "content": [
        {
          "element": "member",
          "meta": {
            "classes": {
              "element": "array",
              "content": [
                {
                  "element": "string",
                  "content": "user"
                }
              ]
            }
          },
          "content": {
            "key": {
              "element": "string",
              "content": "version"
            },
            "value": {
              "element": "string",
              "content": "1.0.0"
            }
          }
        }
      ]
    }
  },
  "content": [ ... ]
}
```

Reply:
I don't think the solution is to have multiple areas which are specific to each API Description format, the point in API Elements is to provide a single cannonical form so that tooling doesn't need to know about the differences between the formats. We are also parsing multiple parts of the Swagger 2 Info Object into other places already, `info.title` goes to the meta.title inside the API Element, `info.description` maps to a `copy` element found within the API Element.

Having separate metadata for API Blueprint, info for Swagger is confusing and doesn't align with those goals. It doesn't provide tooling a standardised way to support multiple API Description formats.

Let's take a step back and look at the different API Description languages and how they offer a version.

| Language      | API Description Version |
|---------------|-------------------------|
| API Blueprint | Unsupported |
| Swagger 2     | `info.version` |
| OAS 3         | `info.version` |
| RAML          | `version` |
| Google Discovery | `version` |

Version is a pretty standardised concept and support across the majority of popular API Description formats. The only outlier is API Blueprint. It doesn't feel like it should be a Swagger/OAS specific feature, the concept of having a version of your API Description Document makes sense and I think it deserves a place in API Elements. It perhaps makes just as much sense as an API title or API Description.

We could stard standardising on some metadata such as the version into the existing metadata section. Although this is currently user-defined and we do not provide semantics here so that would be a change in direction, which may not be a bad idea. One of the problems with the metadata section is that in other languages like API Blueprint all of the metadata are in forced capitals. In API Blueprint we're also mixing the concept of declaring the version of the API Description Format itself (FORMAT: 1A) the term "version" to refer to the version of the format does make sense too, and that is what it is called in other description languages. For example OpenAPI version. It may be confusing that these terms are mixed across API Description languages.

If the version of the API Description Document is called metadata, then why is the title and description not metadata too?

#### Swagger 2

```yaml
swagger: '2.0'
info:
  title: Hello World
  version: 1.0.0
  description: Some Description
```

```json
{
  "element": "category",
  "meta": {
    "title": "Hello World"
  },
  "attributes": {
    "version": "1.0.0"
  },
  "content": [
    {
      "element": "copy",
      "content": "Some Description"
    }
  ]
}
```

#### RAML 1

```yaml
#%RAML 1.0
title: Hello World
version: 1.0.0
description: Some Description
```

#### API Blueprint

```yaml
# Hello World

Some Description
```

@klokane @pksunkara What are your opinions on version and where it belongs?

Reply:
My very first idea was just place `version` to same level like `title` and `description` is. But by spec they both are reserved word for `meta`. Spec does not deny expand other keys in `meta`.
So it seems good for me.

On other side argue specs says:
> `attributes` - SHALL be a set of properties defined by the type of this Element

`version` is specific for root element of API, so maybe it is better idea to place it in attributes.

Reply:
Given more though, I think placing version inside attributes makes the most sense. It is more consistent with other elements.

Reply:
I like your swagger example.